src/node_sqlite.cc

```diff
@@ -128,122 +135,126 @@ inline void THROW_ERR_SQLITE_ERROR(Isolate* isolate, int errcode) {
+                                         Local<Function> fn,
+                                         DatabaseSync* db,
+                                         bool use_bigint_args)
+    : env_(env),
+      fn_(env->isolate(), fn),
+      db_(db),
+      use_bigint_args_(use_bigint_args) {}
+
+UserDefinedFunction::~UserDefinedFunction() {}
+
+void UserDefinedFunction::xFunc(sqlite3_context* ctx,
+                                int argc,
+                                sqlite3_value** argv) {
+  UserDefinedFunction* self =
+      static_cast<UserDefinedFunction*>(sqlite3_user_data(ctx));
+  Environment* env = self->env_;
+  Isolate* isolate = env->isolate();
+  auto recv = Undefined(isolate);
+  auto fn = self->fn_.Get(isolate);
+  LocalVector<Value> js_argv(isolate);
+
+  for (int i = 0; i < argc; ++i) {
+    sqlite3_value* value = argv[i];
+    MaybeLocal<Value> js_val;
+
+    switch (sqlite3_value_type(value)) {
+      case SQLITE_INTEGER: {
+        sqlite3_int64 val = sqlite3_value_int64(value);
+        if (self->use_bigint_args_) {
+          js_val = BigInt::New(isolate, val);
+        } else if (std::abs(val) <= kMaxSafeJsInteger) {
+          js_val = Number::New(isolate, val);
+        } else {
+          self->db_->SetIgnoreNextSQLiteError(true);
+          sqlite3_result_error(ctx, "", 0);
+          THROW_ERR_OUT_OF_RANGE(isolate,
+                                 "Value is too large to be represented as a "
+                                 "JavaScript number: %" PRId64,
+                                 val);
+          return;
         }
-        default:
-          UNREACHABLE("Bad SQLite value");
+        break;
       }
-
-      Local<Value> local;
-      if (!js_val.ToLocal(&local)) {
-        return;
+      case SQLITE_FLOAT:
+        js_val = Number::New(isolate, sqlite3_value_double(value));
+        break;
+      case SQLITE_TEXT: {
+        const char* v =
+            reinterpret_cast<const char*>(sqlite3_value_text(value));
+        js_val = String::NewFromUtf8(isolate, v).As<Value>();
+        break;
       }
-
-      js_argv.emplace_back(local);
+      case SQLITE_NULL:
+        js_val = Null(isolate);
+        break;
+      case SQLITE_BLOB: {
+        size_t size = static_cast<size_t>(sqlite3_value_bytes(value));
+        auto data = reinterpret_cast<const uint8_t*>(sqlite3_value_blob(value));
+        auto store = ArrayBuffer::NewBackingStore(isolate, size);
+        memcpy(store->Data(), data, size);
+        auto ab = ArrayBuffer::New(isolate, std::move(store));
+        js_val = Uint8Array::New(ab, 0, size);
+        break;
+      }
+      default:
+        UNREACHABLE("Bad SQLite value");
     }
 
-    MaybeLocal<Value> retval =
-        fn->Call(env->context(), recv, argc, js_argv.data());
-    Local<Value> result;
-    if (!retval.ToLocal(&result)) {
+    Local<Value> local;
+    if (!js_val.ToLocal(&local)) {
+      self->db_->SetIgnoreNextSQLiteError(true);
```

Comment:
Nit... simply because it may not be obvious for new people coming into this code later, a simple comment might be worthwhile here...
```suggestion
      // Ignore the SQLite error because a v8 error should already be scheduled.
      self->db_->SetIgnoreNextSQLiteError(true);
```